store/cachekv/internal/btree.go

```diff
@@ -21,46 +22,53 @@ var errKeyEmpty = errors.New("key cannot be empty")
 //
 // We choose tidwall/btree over google/btree here because it provides API to implement step iterator directly.
 type BTree struct {
-	tree btree.BTreeG[item]
+	tree *btree.BTreeG[item]
 }
 
 // NewBTree creates a wrapper around `btree.BTreeG`.
-func NewBTree() *BTree {
-	return &BTree{tree: *btree.NewBTreeGOptions(byKeys, btree.Options{
-		Degree: bTreeDegree,
-		// Contract: cachekv store must not be called concurrently
-		NoLocks: true,
+func NewBTree() BTree {
+	return BTree{tree: btree.NewBTreeGOptions(byKeys, btree.Options{
+		Degree:  bTreeDegree,
+		NoLocks: false,
 	})}
 }
 
-func (bt *BTree) Set(key, value []byte) {
+func (bt BTree) Set(key, value []byte) {
 	bt.tree.Set(newItem(key, value))
 }
 
-func (bt *BTree) Get(key []byte) []byte {
+func (bt BTree) Get(key []byte) []byte {
 	i, found := bt.tree.Get(newItem(key, nil))
 	if !found {
 		return nil
 	}
 	return i.value
 }
 
-func (bt *BTree) Delete(key []byte) {
+func (bt BTree) Delete(key []byte) {
 	bt.tree.Delete(newItem(key, nil))
 }
 
-func (bt *BTree) Iterator(start, end []byte) (*memIterator, error) { //nolint:revive
+func (bt BTree) Iterator(start, end []byte) types.Iterator {
 	if (start != nil && len(start) == 0) || (end != nil && len(end) == 0) {
-		return nil, errKeyEmpty
+		panic(errKeyEmpty)
```

Comment:
Why are we panic over error? Ideally, the caller can handle the error.

Reply:
I'm just trying to make it similar to `KVStore` interface, it's usage in the store is side with the parent store, so i try to make their interfaces similar.

Reply:
Yes, but I'd argue those interfaces are bad.

Reply:
agree, fixed.